jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/definition/list/ProcessDefinitionListFilterSettingsManager.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.workbench.pr.client.editors.definition.list;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.enterprise.context.Dependent;
+
+import org.jbpm.workbench.df.client.filter.FilterSettings;
+import org.jbpm.workbench.df.client.filter.FilterSettingsManagerImpl;
+import org.jbpm.workbench.pr.client.resources.i18n.Constants;
+
+import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.*;
+
+@Dependent
+public class ProcessDefinitionListFilterSettingsManager extends FilterSettingsManagerImpl {
+
+    protected static final String TAB_ALL = PROCESS_DEFINITION_LIST_PREFIX + "_0";
```

Comment:
@bxf12315 I dont think there is much value in having a filter that doesnt filter on any attribute. Do we need to keep this?

Reply:
yes, you are right, but if saved filter is empty, here [https://github.com/kiegroup/jbpm-wb/blob/3409fa038b50d70f0fe1d60285e6db57088acdd2/jbpm-wb-common/jbpm-wb-data-filters-client/src/main/java/org/jbpm/workbench/df/client/filter/FilterSettingsManagerImpl.java#L149](url) will throws NPE , so I add default one. I tried to change it, but it brought more bugs.

Reply:
seems like you would need to handle the fact that there are no default filter. @nmirasch based on your recent changes for default filters, would that be acceptable or default filter is now mandatory?

Reply:
@cristianonicolai @bxf12315 yes, we need to add a default 'default filter' here ..https://github.com/kiegroup/jbpm-wb/blob/3409fa038b50d70f0fe1d60285e6db57088acdd2/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListFilterSettingsManager.java#L63 setting the key to 'getDefaultFilterSettingsKey()'. That would be the one used in case no other default filter is checked, and yes, define a default filter is mandatory.